templates/network-policies.yaml

```diff
@@ -0,0 +1,49 @@
+{{- if .Values.networkPolicies.enabled }}
+---
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-default-ingress
+  labels:
+    app.kubernetes.io/name: {{ include "temporal.name" . }}
+    helm.sh/chart: {{ include "temporal.chart" . }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/version: {{ .Chart.AppVersion | replace "+" "_" }}
+    app.kubernetes.io/part-of: {{ .Chart.Name }}
+spec:
+  podSelector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "temporal.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  ingress:
+    - from:
+        - podSelector:
+            matchLabels:
+              app.kubernetes.io/name: {{ include "temporal.name" . }}
+              app.kubernetes.io/instance: {{ .Release.Name }}
+      ports:
+        {{- range $service := (list "frontend" "history" "matching" "worker") }}
+        {{- $serviceValues := index $.Values.server $service }}
+        - protocol: TCP
+          port: {{ $serviceValues.service.port }}
+        {{- end }}
```

Comment:
does this work if fontend, history, matching and worker all share the same port?